e2e_playwright/st_segmented_control.py

```diff
@@ -0,0 +1,182 @@
+            "Foobar",
+            "🧰 General widgets",
+        ]
+    else:
+        st.session_state.default_segmented_control_options = []
+
+    default = st.session_state.default_segmented_control_options
+
+    selection = st.segmented_control(
+        "select some options",
+        [
+            ":material/star: Hello there!",
+            "Foobar",
+            "Icon in the end: :material/rocket:",
+            ":material/thumb_up: Hello again!",
+            "🧰 General widgets",
+            "📊 Charts",
+            "🌇 Images",
+            "🎥 Video",
+            "📝 Text",
+            "This is a very long text 📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝, yes, long long long long text",
+        ],
+        key="segmented_control",
+        selection_mode="multiple",
+        default=default,
+        help="You can choose multiple options",
+    )
+
+    st.write(f"Multi selection: {selection}")
+
+
+st.header("Single Select - Segmented Control", anchor="single-select-segmented-control")
+with st.echo(code_location="below"):
+    selection = st.segmented_control(
+        "select an option",
+        [
+            ":material/star: Hello there!",
+            "Foobar",
+            "Icon in the end: :material/rocket:",
+        ],
+        selection_mode="single",
+    )
+    st.write(f"Single selection: {selection}")
+
+option_to_icon_map = {
+    0: ":material/add:",
+    1: ":material/zoom_in:",
+    2: ":material/zoom_out:",
+    3: ":material/zoom_out_map:",
+}
+
+st.header(
+    "Icon-only button group - Segmented Control",
+    anchor="icon-only-button-group-segmented-control",
+)
+with st.echo(code_location="below"):
+    selection = st.segmented_control(
+        "select an icon",
+        options=[0, 1, 2, 3],
+        format_func=lambda option: option_to_icon_map[option],
+        selection_mode="single",
+    )
+    st.write(f"Single icon selection: {selection}")
+
+
+st.header(
+    "on_change callback - Segmented Control",
+    anchor="on-change-callback-segmented-control",
+)
+with st.echo(code_location="below"):
+    st.segmented_control(
+        "Emotions",
+        ["Joy", "Sadness", "Anger", "Disgust"],
+        key="segmented_control_on_change",
+        on_change=lambda: st.write(
+            f"on_change selection: {st.session_state.segmented_control_on_change}"
+        ),
+    )
+
+
+st.header("Disabled - Segmented Control", anchor="disabled-segmented-control")
+with st.echo(code_location="below"):
+    selection = st.segmented_control(
+        "Emotions",
+        ["Joy", "Sadness", "Anger", "Disgust"],
+        key="segmented_control_disabled",
+        disabled=True,
+    )
+    st.write("segmented-control-disabled:", str(selection))
+
+
+st.header("Segmented Control in form", anchor="segmented-control-in-form")
+with st.echo(code_location="below"):
+    with st.form(key="my_form", clear_on_submit=True):
+        selection = st.segmented_control(
+            "Emotions",
```

Comment:
nit: I think it's better to give all widgets unique and descriptive labels, e.g.: "segmented control (disabled)" and change the test script to get the widget based on the label instead of the index. Most of our test scripts are using indices to get an element, but that's a bit of a bad pattern (see [this technical issue item](https://www.notion.so/snowflake-corp/In-e2e-tests-using-nth-n-to-find-the-right-element-for-that-test-seems-like-a-fragile-way-to-set--10c7170bb41680738d71c235e87d30d5))